doc/architecture/decisions/0014-p2p-ipc-abstraction.md

```diff
@@ -0,0 +1,32 @@
+# 14. P2P - IPC Abstraction
+
+Date: 2018-07-19
+
+## Status
+
+Proposed
+
+## Context
+
+It's frustrating to admit, but I'm still pretty slow at writing rust code. Oh, I can hack something together quickly (gogo `unwrap()/clone()`), but when I sit down to write efficient production code, I get bogged down in cyclic lifetime reference issues or any number of other arcane rustisisms.
```

Comment:
this doesn't seem like relevant context for an architecture design decision, it's just a comment about our personal situations at this point in time